perf-tests/src/gremlin_package_generator.py

```diff
@@ -0,0 +1,90 @@
+"""Class that generates ecosystem+package and ecosystem+package+version tuples."""
+
+
+class GremlinPackageGenerator:
+    """Class that generates ecosystem+package and ecosystem+package+version tuples."""
+
+    PACKAGES = {
+        "npm": {
+            "sequence": ["3.0.0"]
+        },
+        "maven": {
+            "io.vertx:vertx-core": ["3.4.0"]
+        },
+        "pypi": {
+            "clojure_py": ["0.2.0", "0.2.1", "0.2.2", "0.2.3", "0.2.4"],
+            "ansicolors": ["1.0", "1.0.1", "1.0.2", "1.1.5", "1.1.6", "1.1.7",
+                           "1.1.8"],
+        }
+    }
+
+    @staticmethod
+    def generate_ecosystem_package(ecosystem, packages):
+        """Generate sequence of tuples containing ecosystem+package name pairs."""
+        for package in packages:
+            yield (ecosystem, package)
+
+    @staticmethod
+    def generate_ecosystem_package_version(ecosystem, packages):
+        """Generate sequence of triples containing ecosystem+package name pairs."""
+        for package, versions in packages.items():
+            for version in versions:
+                yield (ecosystem, package, version)
+
+    @staticmethod
+    def package_generator():
+        """Provide generator for package specifications across all ecosystems."""
+        while True:
```

Comment:
Do we need the `while True:` here and in other functions ?
I've been looking at [this](https://stackoverflow.com/questions/9708902/in-practice-what-are-the-main-uses-for-the-new-yield-from-syntax-in-python-3) and it seems that it's in general used when you're sending data to a generator (coroutine).

Reply:
Thanks @jpopelka for review. I've added a few comments why the endless loop is used there.